src/pycram/datastructures/world.py

```diff
@@ -1080,84 +1571,144 @@ def __enter__(self):
         This method is called when entering the with block, it will set the current world to the prospection world
         """
         if not World.current_world.is_prospection_world:
-            time.sleep(self.WAIT_TIME_FOR_ADDING_QUEUE * World.current_world.simulation_time_step)
-            # blocks until the adding queue is ready
-            World.current_world.world_sync.add_obj_queue.join()
-            self.sync_worlds()
```

Comment:
It is done when entering the prospection world, but the queue here is not needed because for example if you add an object and then remove it then when syncing using the queue it will add the object and then remove it which is an unnecessary step. instead once With UseProspectionWorld is used, the world will be compared at the current state and only the non-existing objects in prospection will be added, while objects in the prospection world that are not in the main world will be removed. this removes the need for managing a queue.